File: services/agent_service.py
"""
agent_service.py – Gestione agenti AI: creazione, assegnazione, disattivazione.
"""
from models.agent import Agent
import uuid
from datetime import datetime

class AgentService:
    def __init__(self, db):
        self.db = db

    def create_agent(self, owner_id: str, agent_type: str, skills=None, meta=None):
        agent = Agent(
            id=str(uuid.uuid4()),
            owner_id=owner_id,
            agent_type=agent_type,
            skills=skills or [],
            status="active",
            created_at=datetime.utcnow(),
            meta=meta or {}
        )
        self.db.save(agent)
        return agent

    def get_agent(self, agent_id: str):
        return self.db.get(Agent, agent_id)

    def deactivate_agent(self, agent_id: str):
        agent = self.get_agent(agent_id)
        if agent:
            agent.status = "inactive"
            self.db.save(agent)
        return agent

File: services/agentService.js
// /services/agentService.js

const db = require("../db");
const { v4: uuidv4 } = require("uuid");

class AgentService {
  async createAgent({ ownerId, type, config }) {
    const agent = {
      id: uuidv4(),
      ownerId,
      type,
      config,
      createdAt: new Date(),
      updatedAt: new Date(),
      status: "active",
    };
    await db.insert("agents", agent);
    return agent;
  }

  async getAgentById(id) {
    return db.findOne("agents", { id });
  }

  async updateAgent(id, updates) {
    updates.updatedAt = new Date();
    await db.update("agents", id, updates);
    return this.getAgentById(id);
  }

  async deleteAgent(id) {
    await db.delete("agents", id);
  }
}

module.exports = new AgentService();
